### PR TITLE
Merge mpi kokkos

### LINKED ISSUE #177: Merge NimbleSM_MPI with NimbleSM_Kokkos

### PR DESCRIPTION
**Pull request by uhetmaniuk**

### CODE REVIEW

File: src/nimble_contact_manager.cc [hunk 1]
```diff
@@ -618,971 +394,1343 @@ namespace nimble {
...
+                elem_data_labels_for_output, elem_data_for_output,
+                derived_elem_data_labels, derived_elem_data);
+}
+
+
+//
+// Protected functions
+//
+
+void ContactManager::ApplyDisplacements(const double *displacement) {
```
Comment (rjones-sandia) [inline src/nimble_contact_manager.cc:745]: thanks, I really happy to see that this object is now composed of shorter functions that (more or less) do not scroll off the screen

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -143,19 +143,19 @@ ENDIF()
 
 list(APPEND lstTests_ArborX_A
 #        brick_with_fiber_periodic
-        brick_with_fibers
```
Comment (rjones-sandia) [inline test/CMakeLists.txt:146]: are some of the tests breaking with this code change?

**Review by rjones-sandia — APPROVED** [review]
this is a big step, nice work

- it looks like a few tests were commented out, I'm assuming I either misunderstood this or this is an interim state

File: src/nimble_contact_manager.cc (hunk 1 above)
Comment (uhetmaniuk) [reply src/nimble_contact_manager.cc:745]: @rjones-sandia 
Great!
I tried to keep the modifications for `nimble_contact_manager.cc` as limited as possible.
If I remembered correctly, we will update the `ContactManager` hierarchy.
So I focused on making everything work.

File: test/CMakeLists.txt (hunk 2 above)
Comment (uhetmaniuk) [reply test/CMakeLists.txt:146]: @rjones-sandia 
You are correct that it is an intermediate step.
The problem is the following. The current CI uses explicitly:
- the executable `NimbleSM_ArborX` (-> I solved that);
- a distinction between the output files with `ArborX` and the ones with `Kokkos` (no contact).

So the uncommented tests are the ones with `ArborX`.

I will ask a question about the naming of output files at the meeting: with only one executable, do we want only one naming scheme for the output?

**Pull request merged**